wp-content/themes/core/pcss/components/_dialog.pcss

```diff
@@ -0,0 +1,87 @@
+/* -----------------------------------------------------------------------------
+ *
+ * Component: Dialog
+ *
+ * ----------------------------------------------------------------------------- */
+
+.c-dialog[aria-hidden='true'] {
+	display: none;
+}
+
+.c-dialog__overlay {
+	background-color: var(--color-goethite);
+	bottom: 0;
+	color: var(--color-white);
+	left: 0;
+	overflow: scroll;
+	position: fixed;
+	right: 0;
+	top: 0;
+	width: 100%;
+	z-index: 2;
+
+	.admin-bar & {
+		top: var(--spacer-wpadmin-mobile);
+
+		@media (--viewport-wpadmin) {
+			top: var(--spacer-wpadmin-desktop);
+		}
+	}
+}
+
+.c-dialog__header {
+	background-color: var(--color-goethite);
+	border-bottom: 1px solid var(--color-white);
+	margin-bottom: 47px;
+	padding: 20px;
+	position: fixed;
```

Comment:
See above, going fixed on this in particular is fragile, depending on header title or whatever content may end up in the header from a height perspective. We can remove the fixed handling once you update the fixed item above and for now just allow the whole overlay to scroll, we'll do a refinement pass after this first pass on common panels). We could also potentially try to handle this as a flex situation where the header remains stuck and content fills up whatever area is left if that wouldn't take much time.